libfwupdplugin/fu-udev-device.h

```diff
@@ -55,6 +55,56 @@ typedef enum {
 	FU_UDEV_DEVICE_FLAG_LAST
 } FuUdevDeviceFlags;
 
+/**
+ * FuPciBaseClass:
+ * @FU_PCI_BASE_CLASS_OLD: Device from before classes were defined
+ * @FU_PCI_BASE_CLASS_MASS_STORAGE: Mass Storage Controller
+ * @FU_PCI_BASE_CLASS_NETWORK: Network controller
+ * @FU_PCI_BASE_CLASS_DISPLAY: Display controller
+ * @FU_PCI_BASE_CLASS_MULTIMEDIA: Multimedia controller
+ * @FU_PCI_BASE_CLASS_MEMORY: Memory controller
+ * @FU_PCI_BASE_CLASS_BRIDGE: Bridge
+ * @FU_PCI_BASE_CLASS_SIMPLE_COMMUNICATION: Simple communications controller
+ * @FU_PCI_BASE_CLASS_BASE: Base system peripheral
+ * @FU_PCI_BASE_CLASS_INPUT: Input device
+ * @FU_PCI_BASE_CLASS_DOCKING: Docking station
+ * @FU_PCI_BASE_CLASS_PROCESSORS: Processor
+ * @FU_PCI_BASE_CLASS_SERIAL_BUS: Serial bus controller
+ * @FU_PCI_BASE_CLASS_WIRELESS: Wireless controller
+ * @FU_PCI_BASE_CLASS_INTELLIGENT_IO: Intelligent IO controller
+ * @FU_PCI_BASE_CLASS_SATELLITE: Satellite controller
+ * @FU_PCI_BASE_CLASS_ENCRYPTION: Encryption/Decryption controller
+ * @FU_PCI_BASE_CLASS_SIGNAL_PROCESSING: Data acquisition and signal processing controller
+ * @FU_PCI_BASE_CLASS_ACCELERATOR: Processing accelerator
+ * @FU_PCI_BASE_CLASS_NON_ESSENTIAL: Non-essential instrumentation
+ * @FU_PCI_BASE_CLASS_UNDEFINED: Device doesn't fit any defined class
+ *
+ * PCI base class types returned by fu_udev_device_get_class().
+ **/
+typedef enum {
+	FU_PCI_BASE_CLASS_OLD,
+	FU_PCI_BASE_CLASS_MASS_STORAGE,
+	FU_PCI_BASE_CLASS_NETWORK,
+	FU_PCI_BASE_CLASS_DISPLAY,
+	FU_PCI_BASE_CLASS_MULTIMEDIA,
+	FU_PCI_BASE_CLASS_MEMORY,
+	FU_PCI_BASE_CLASS_BRIDGE,
+	FU_PCI_BASE_CLASS_SIMPLE_COMMUNICATION,
+	FU_PCI_BASE_CLASS_BASE,
+	FU_PCI_BASE_CLASS_INPUT,
+	FU_PCI_BASE_CLASS_DOCKING,
+	FU_PCI_BASE_CLASS_PROCESSORS,
+	FU_PCI_BASE_CLASS_SERIAL_BUS,
+	FU_PCI_BASE_CLASS_WIRELESS,
+	FU_PCI_BASE_CLASS_INTELLIGENT_IO,
+	FU_PCI_BASE_CLASS_SATELLITE,
+	FU_PCI_BASE_CLASS_ENCRYPTION,
+	FU_PCI_BASE_CLASS_SIGNAL_PROCESSING,
+	FU_PCI_BASE_CLASS_ACCELERATOR,
+	FU_PCI_BASE_CLASS_NON_ESSENTIAL,
+	FU_PCI_BASE_CLASS_UNDEFINED = 0xff
+} FuPciBaseClass;
```

Comment:
Not sure about the prefix; how about `FuUdevDevicePciBase` -- using _Class_ is a recipe for disaster I think

Reply:
How about CLS? I'm worried it really is a class and we want to indicate that